pkg/asset/installconfig/password.go

```diff
@@ -0,0 +1,63 @@
+package installconfig
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/openshift/installer/pkg/asset"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+var (
+	passwordPrompt = fmt.Sprintf("Enter password (6-16 characters):")
+	reEnterPrompt  = fmt.Sprintf("Re-enter password:")
+)
+
+// password is an asset that queries the user for the password to the cluster
+//
+// Contents[0] is the actual string form of the password
+type password struct {
+	InputReader *bufio.Reader
+}
+
+var _ asset.Asset = (*password)(nil)
+
+// Dependencies returns no dependencies.
+func (a *password) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate queries for input from the user.
+func (a *password) Generate(map[asset.Asset]*asset.State) (*asset.State, error) {
+	password, err := a.queryUserForPassword()
+	return assetStateForStringContents(password), err
+}
+
+// queryUserForPassword for prompts Stdin for password without echoing anything on screen
+// TODO: mask the password characters
+func (a *password) queryUserForPassword() (string, error) {
+	for {
+		fmt.Printf(passwordPrompt)
+		input, err := terminal.ReadPassword(int(os.Stdin.Fd()))
+		if err != nil {
+			fmt.Printf("error in getting pasword %v", err)
+			return "", err
+		}
+		fmt.Println()
+		if len(input) < 6 || len(input) > 16 {
+			fmt.Println("Password length should be 6-16 characters.")
+			continue
+		}
+		fmt.Printf(reEnterPrompt)
+		confirmInput, err := terminal.ReadPassword(int(os.Stdin.Fd())) //reEnterPrompt)
+		if err != nil {
+			return "", err
```

Comment:
"failed to read password: %v"

Reply:
Why print this out here instead of returning the error?

Reply:
Since we were just interacting with the user anyway. The returned error may just be logged instead. Remove it?

Reply:
> Since we were just interacting with the user anyway. The returned error may just be logged instead.

I'd much rather bubble these up and return them as errors, which are eventually logged [here][1].  I'm fine wrapping the error or not, but don't want to log directly here.

[1]: https://github.com/openshift/installer/blob/00cb2f115b1e57e7106422f4848e2bca7cd02e30/cmd/openshift-install/main.go#L49